### PR TITLE
Problem: Wasting resource to retry on network error

### LINKED ISSUE #474: Problem: Wasting resource to retry on network error

### PR DESCRIPTION
**Pull request by calvinaco**
Solution: (Fix #474) Use exponential backoff to retry HTTP request on sync

### CODE REVIEW

File: infrastructure/feed/chain/blockheighttracker.go [hunk 1]
```diff
@@ -45,27 +48,37 @@ func NewBlockHeightTracker(logger applogger.Logger, client tendermint.Client) *B
...
+			tracker.logger.Infof("updated chain latest block height: %d", height)
+			<-time.After(tracker.pollingInterval)
+			return nil
+		}
+		notifyFn := func(opErr error, backoffDuration time.Duration) {
+			tracker.logger.Errorf("retrying in %s: %v", backoffDuration.String(), opErr)
+		}
+		neverStopExponentialBackoff := backoff.NewExponentialBackOff()
+		neverStopExponentialBackoff.MaxElapsedTime = 0
+		neverStopExponentialBackoff.MaxInterval = 15 * time.Minute
```
Comment (davcrypto) [inline infrastructure/feed/chain/blockheighttracker.go:78]: shall we put this as const on the top for easier configuration?

File: cmd/chain-indexing/syncmanager.go [hunk 2]
```diff
@@ -206,18 +214,40 @@ func (manager *SyncManager) Run() error {
...
+				"retrying in %s: %v", backoffDuration.String(), opErr,
+			)
 		}
 
-		select {
-		case <-manager.shouldSyncCh:
-		case <-time.After(manager.pollingInterval):
+		neverStopExponentialBackoff := backoff.NewExponentialBackOff()
+		neverStopExponentialBackoff.MaxElapsedTime = 0
+		neverStopExponentialBackoff.MaxInterval = 15 * time.Minute
```
Comment (davcrypto) [inline cmd/chain-indexing/syncmanager.go:244]: shall we put this as const on the top for easier configuration?

**Comment by davcrypto** [comment]
@calvinaco just a minor comment, overall LGTM

File: infrastructure/feed/chain/blockheighttracker.go (hunk 1 above)
Comment (calvinaco) [reply infrastructure/feed/chain/blockheighttracker.go]: Good point, done

File: cmd/chain-indexing/syncmanager.go (hunk 2 above)
Comment (calvinaco) [reply cmd/chain-indexing/syncmanager.go]: Good point, done

**Pull request merged**